Add React Context for state management

ishidaamna opened the pull request:
1. Created a new folder named 'store' and a new file named 'shopping-cart-context.jsx' in it.

2. Created a CartContext using createContext.

3. Wrapped CartContext around the Header and Shop components.

ishidaamna committed:
Add React Context for state management

1. Created a new folder named 'store' and a new file named 'shopping-cart-context.jsx' in it.

2. Created a CartContext using createContext.

3. Wrapped it around the Header and Shop components.

abdulwahabone commented on src/App.jsx:
```diff
@@ -3,6 +3,7 @@ import Product from "./components/Product.jsx";
 import Header from "./components/Header.jsx";
 import Shop from "./components/Shop.jsx";
 import { DUMMY_PRODUCTS } from "./dummy-products.js";
+import { CartContext } from "./store/shopping-cart-context.jsx";
```
You dont need to use `.jsx` at the end

ishidaamna replied:
> You dont need to use `.jsx` at the end

Just on line 6 or all of the imports?

ishidaamna committed:
This PR consumes the context using the useContext hook

1. Got rid of the items prop from the Cart component.

2. Added value to the Provider component.

ishidaamna committed:
Added gitignore

1. Added gitignore file.

2. Formated document and made the code cleaner.

ishidaamna committed:
Migrating to context API

1. fully migrated the project to Context API.

2. Formatted document

3. Got ride of unnecessery props.

abdulwahabone reviewed (approved):
Look sgood
